pkg/search/auth/auth.go

```diff
@@ -0,0 +1,33 @@
+package auth
+
+import (
+	"context"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+)
+
+// WithAuthFilter filters searcher with authorization check for the specified resource
+func WithAuthFilter(searcher search.Searcher, resource permissions.ResourceHandle) search.Searcher {
```

Comment:
We didn't already have something like this?

Reply:
Nope.

Reply:
I don't think this is necessary. See above, but there are sac helpers and other auth helpers that do this